x-pack/plugin/core/src/main/resources/ecs-template.yml

```diff
@@ -0,0 +1,222 @@
+    # any field that shows up after the first 1000 fields is kept in source
+    # but does not show up as runtime for index field
+    #index.mapping.total_fields.soft_limit: 1000
+
+    # Ignores any malformed data. Data will still be persisted in _source
+    index.mapping.ignore_malformed": true
+
+  mappings:
+
+    # Any field that is discovered is mapped as a runtime field
+    dynamic: runtime
+
+    # DISCUSS: What are the pros / cons of having this enabled?
+    date_detection: false
+
+    ### All the properties which are set in each template
+    properties:
+
+      # This ensures all subobjects are flattened and prevents and object / keyword conflicts
+      # As subobject is set to false, no objects like host, error or similar are mapped.
+      subobject: false
+
+      # @timestamp is the only field that MUST exist.
+      "@timestamp":
+        type: date
+        ignore_malformed: false
+
+      # The data stream properties are required in any index following the data stream naming scheme
+      # Discuss: If there is an ecs: enabled flag it can also be used in indices so this would have to be skipped
+      # These fields might not even be needed in this template
+      data_stream.type:
+        type: constant_keyword
+        #value: logs
+      data_stream.dataset:
+        type: constant_keyword
+      data_stream.namespace:
+        type: constant_keyword
+
+
+    # To ensure no empty mappings are in an index, everything except the properties above are mapped as dynamic templates
+    # The dynamic mappings here are a best effort to map ECS core fields: https://github.com/elastic/ecs/blob/8.4/generated/csv/fields.csv
+    # As most fields are keyword, these will be automatically be matched correctly. Only core fields were taken into account.
+    # DISCUSS: Should @timestamp and data_stream.* also be moved under dynamic templates?
+    dynamic_templates:
+
+      ### Index mappings
+
+      # Message field is always index
+      - message:
+          path_match: message
+          mapping:
+            type: match_only_text
+
+      ### Runtime mappings
+
+      # Matching all string fields by default to keyword
+      # TODO: Does first / last match win in the list?
+      - strings_as_keyword:
+          match_mapping_type: string
+          runtime:
+            type: keyword
+            # Do we need this for runtime?
+            ignore_above: 1024
+
+      # Matching all long fields
+      # This matches the following fields: bytes, packets, port, duration, severity, pid
+      # No path matching is needed as it is assumed these fields are already in the json doc a long
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "7" as string comes in, will it still be mapped correctly?
+      - long_as_long:
+          match_mapping_type: long
+          runtime:
+            type: long
```

Comment:
I would also think that this is not needed as it is the default behaviour with dynamic:runtime

Reply:
another option to prevent type conflicts would be to have a named based convention instead of type based: list the field names and their mappings, then even if they come in as strings they will be mapped as long. Another question is if these fields are only a few and appear often in documents, is it worth mapping them manually. I know I am going a bit back and forth on this, we need to find a good balance :)

Reply:
I'm by now leaning towards matching based on names. Reason is to ensure `"7"` and `7` are both matched to long. At least my understanding is this required as otherwise `"7"` would not be a long.

Reply:
Moved to fully name based matching